components/engine_rocks/src/ttl_properties.rs

```diff
@@ -0,0 +1,159 @@
+// Copyright 2021 TiKV Project Authors. Licensed under Apache-2.0.
+
+use std::collections::HashMap;
+
+use crate::{RocksEngine, UserProperties};
+use engine_traits::util::get_expire_ts;
+use engine_traits::{
+    DecodeProperties, Range, Result, TtlProperties, TtlPropertiesExt, TableProperties,
+    TablePropertiesCollection, TablePropertiesExt,
+};
+use rocksdb::{DBEntryType, TablePropertiesCollector, TablePropertiesCollectorFactory};
+
+const PROP_MAX_EXPIRE_TS: &str = "tikv.max_expire_ts";
+const PROP_MIN_EXPIRE_TS: &str = "tikv.min_expire_ts";
+
+pub struct RocksTtlProperties;
+
+impl RocksTtlProperties {
+    pub fn encode(ttl_props: &TtlProperties) -> UserProperties {
+        let mut props = UserProperties::new();
+        props.encode_u64(PROP_MAX_EXPIRE_TS, ttl_props.max_expire_ts);
+        props.encode_u64(PROP_MIN_EXPIRE_TS, ttl_props.min_expire_ts);
+        props
+    }
+
+    pub fn decode<T: DecodeProperties>(props: &T) -> Result<TtlProperties> {
+        let mut res = TtlProperties::default();
+        res.max_expire_ts = props.decode_u64(PROP_MAX_EXPIRE_TS)?;
+        res.min_expire_ts = props.decode_u64(PROP_MIN_EXPIRE_TS)?;
+        Ok(res)
+    }
+}
+
+impl TtlPropertiesExt for RocksEngine {
```

Comment:
Seems this logic is same for `MvccProperties` and `TtlProperties`. Just a question, is it possible to merge them like
```rust
trait PropertiesExt {
    type Properties = X;
    fn get_range_properties_cf(...) -> Reuslt<Vec<(String, Self::Properties)>>;
}
```

Reply:
Seems yes. I think we can leave it for engine traits' future work.